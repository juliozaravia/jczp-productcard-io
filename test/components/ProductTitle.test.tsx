import React from 'react';
import renderer from 'react-test-renderer';

import { ProductTitle, ProductCard } from '../..';

const product1 = {
  id: '1',
  title: 'Coffee Mug - No image',
};

describe('ProductTitle', () => {
  test('It should show the component correctly', () => {
    const wrapper = renderer.create(<ProductTitle title="Custom product" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('It should show the component with the product name', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>{() => <ProductTitle />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
