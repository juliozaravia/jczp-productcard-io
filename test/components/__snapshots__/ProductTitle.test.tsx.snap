// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle It should show the component correctly 1`] = `
<span
  className="styles-module_productDescription__Ariub undefined"
>
  Custom product
</span>
`;

exports[`ProductTitle It should show the component with the product name 1`] = `
<div
  className="styles-module_productCard__oaIVo undefined"
>
  <span
    className="styles-module_productDescription__Ariub undefined"
  >
    Coffee Mug - No image
  </span>
</div>
`;
